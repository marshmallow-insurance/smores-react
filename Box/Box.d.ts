import { FlexProps } from '../utils/flex';
import { MeasureProps } from '../utils/measure';
import { MarginProps, PaddingProps } from '../utils/space';
export type BoxProps = MarginProps & PaddingProps & FlexProps & MeasureProps;
export declare const Box: import("styled-components").StyledComponent<"div", any, MarginProps & PaddingProps & FlexProps & MeasureProps, never>;
