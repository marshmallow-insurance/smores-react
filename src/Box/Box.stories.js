import React from 'react';
import {storiesOf} from '@storybook/react';

import {text, boolean} from '@storybook/addon-knobs';

import {Box} from './Box';

storiesOf('Box', module).addWithJSX('default', () => (
  <Box
    flex={boolean('flex', true)}
    direction={text('direction', 'row')}
    justifyContent={text('justifyContent', 'space-between')}
    alignItems={text('alignItems', 'center')}
    m={text('margin', '16px')}
    p={text('padding', '16px')}
  >
    Some lovely content.
  </Box>
));
