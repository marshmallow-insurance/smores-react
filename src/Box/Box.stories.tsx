import React from 'react'
import { Box, BoxProps } from './Box'

export default {
  title: 'Box',
  component: Box,
}

const Template = (props: BoxProps) => <Box {...props}>Some lovely content.</Box>

export const Default = Template.bind({})

Default.args = {
  flex: true,
  direction: 'row',
  justifyContent: 'flex-start',
  alignItems: 'flex-starts',
  m: '16px',
  p: '16px',
}

Default.argTypes = {
  theme: { table: { disable: true } },
  as: { table: { disable: true } },
  forwardedAs: { table: { disable: true } },
}
