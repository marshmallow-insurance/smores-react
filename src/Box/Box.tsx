import styled from 'styled-components';

import {flex, FlexProps} from '../utils/flex';
import {measure, MeasureProps} from '../utils/measure';
import {space, SpaceProps} from '../utils/space';

interface IBox extends SpaceProps, FlexProps, MeasureProps {}

export const Box = styled.div<IBox>`
  ${space};
  ${flex};
  ${measure};
`;
