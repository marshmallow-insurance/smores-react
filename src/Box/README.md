# Box

The `<Box />` component is a replacement for the commonly used `<div>` tag. It inherits the `flex`,`space` and `measure` utilities to that it can take inline props for flex-box, margin, padding, position, width and height.

## Available Props

### Flex

| Description     | Prop           | Type     |                                Options                                 |
| --------------- | -------------- | -------- | :--------------------------------------------------------------------: |
| flex-direction  | direction      | optional |                column, column-reverse, row, row-reverse                |
| flex-wrap       | wrap           | optional |                       nowrap, wrap, wrap-reverse                       |
| flex            | flex           | optional |                                boolean                                 |
| flex-flow       | flow           | optional |                                 string                                 |
| justify-content | justifyContent | optional | center, flex-end, flex-start, space-around,space-between, space-evenly |
| align-items     | alignItems     | optional |           baseline, center,  flex-end,  flex-start,  stretch           |
| align-content   | alignContent   | optional |   center, flex-end, flex-start, space-around, space-between, stretch   |

### Space

| Description     | Prop | Type     |      Options      |
| --------------- | ---- | -------- | :---------------: |
| margin          | m    | optional | string, in pixels |
| margin x-axis   | mx   | optional | string, in pixels |
| margin y-axis   | my   | optional | string, in pixels |
| margin-left     | ml   | optional | string, in pixels |
| margin-right    | mr   | optional | string, in pixels |
| margin-top      | mt   | optional | string, in pixels |
| margin-bottom   | mb   | optional | string, in pixels |
| padding         | p    | optional | string, in pixels |
| padding x-axis  | px   | optional | string, in pixels |
| padding  y-axis | py   | optional | string, in pixels |
| padding-left    | pl   | optional | string, in pixels |
| padding-right   | pr   | optional | string, in pixels |
| padding-top     | pt   | optional | string, in pixels |
| padding-bottom  | pb   | optional | string, in pixels |

### Measure

| Description | Prop      | Type     |      Options      |
| ----------- | --------- | -------- | :---------------: |
| width       | width     | optional | string, in pixels |
| max-width   | maxWidth  | optional | string, in pixels |
| min-width   | minWidth  | optional | string, in pixels |
| height      | height    | optional | string, in pixels |
| max-height  | maxHeight | optional | string, in pixels |
| min-height  | minHeight | optional | string, in pixels |
