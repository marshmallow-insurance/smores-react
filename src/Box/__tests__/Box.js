import React from 'react';
import {render} from '@testing-library/react';
import 'jest-styled-components';

import {Box} from '../Box';

test('renders', () => {
  const {container} = render(
    <Box pb="16px" mt="32px" flex direction="column" />,
  );
  expect(container.firstChild).toMatchSnapshot();
});
