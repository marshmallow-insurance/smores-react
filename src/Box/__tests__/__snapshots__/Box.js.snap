// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:0px) {
  .c0 {
    margin-top: 32px;
  }
}

@media (min-width:0px) {
  .c0 {
    padding-bottom: 16px;
  }
}

@media (min-width:0px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div
  class="c0"
  direction="column"
/>
`;
